report-ng/app/src/components/timings/test-timings/test-timings.ts

```diff
@@ -0,0 +1,373 @@
+                    }
+
+                    return ""; // Return an empty string if no data points are hovered on
+                }.bind(this), // Binding the current context to the formatter function to access this._durationOptions
+            },
+            xAxis: {
+                type: 'category',
+                data: this._labels,
+                name: 'duration'
+            },
+            yAxis: {
+                type: 'value',
+                minInterval: 1, //allows only integer values
+                name: 'number of test cases'
+            },
+            series: [
+                {
+                    data: this._data,
+                    type: 'bar',
+                    itemStyle: {
+                        color: '#6897EA',
+                    },
+                }
+            ]
+        };
+    }
+
+    private _highlightData() {
+        if (this._bars.length > 0) {
+            const dataIndex = this._bars.findIndex(value => value.methodList.find(value => value.id === this._methodId));
+            this._option.series[0].data = this._data.map((item, index) => {
+                return {
+                    value: item,
+                    itemStyle: {
+                        color: (index === dataIndex) ? '#6897EA' : '#c8d4f4'
+                    }
+                };
+            });
+            this._chart.setOption(this._option)
+        }
+    }
+
+    private _prepareData(methods: ITestDurationMethod[]) {
+        const durations = methods.map(method => method.duration);
+        this._calculateDurationAxis(durations);
+
+        this._bars = this._sectionValues.map((sectionValue, i) => {
+            const methodList = methods.reduce((list, method) => {
+                if (method.duration <= sectionValue && (method.duration > this._sectionValues[i - 1] || this._sectionValues[i - 1] === undefined)) {
+                    list.push(method);
+                }
+                return list;
+            }, []);
+
+            const bar: IDurationBar = {
+                durationAmount: methodList.length,
+                methodList: methodList,
+                label: this._labels[i]
+            };
+            return bar;
+        });
+
+        this._data = this._bars.map(bar => bar.durationAmount);
+    }
+
+    private _calculateDurationAxis(durations: number[]) {
+        this._rangeNum = this.queryParams.rangeNum;
+        durations.sort((a, b) => a - b)
+
+        let maxDuration = durations[durations.length - 1];
+        const remainder = maxDuration % this._rangeNum;
+        maxDuration = remainder === 0 ? maxDuration : maxDuration + (this._rangeNum - remainder);
+
+        const sectionRange = maxDuration / this._rangeNum;
+
+        const resultDurations: number[] = Array(this._rangeNum).fill(0);
+        const resultSections: string[] = Array(this._rangeNum).fill("");
+
+        // calculate ranges
+        for (let i = 0; i < this._rangeNum; i++) {
```

Comment:
It's confusing that you need an integer using `parseInt` in `queryParams`, but all other values are strings like in `_rangeNum` are strings, except here, where you copy integer `queryParams.rangeNum` to `_rangeNum`.

* Please fix the type of `private _rangeNum:string`.
* Do not `parseInt` anymore.
* Only parse in the `for` loop like
```typescript
for (let i = 0; i < parseInt(this._rangeNum); i++)
```

Reply:
I agree, the type mixup is not ideal, yet it was the only working solution I found since the options for the select menu for the range options need to be strings. That is why i set `this._rangeNum = '10'` in the `activate()` method. If i only use `parseInt()` in the for loop i cannot use `this._rangeNum` as value for calculation above in `_calculateDurationAxis`. 
I would have to declare a new integer variable in the beginning and set it once to the int value of the `this._rangeNum` string value. Would that be better?

E.g.:
```
private _calculateDurationAxis(durations: number[]) {
        this._rangeNum = this.queryParams.rangeNum;
        const rangeNumInt = parseInt(this._rangeNum)
        durations.sort((a, b) => a - b)

        let maxDuration = durations[durations.length - 1];
        const remainder = maxDuration % rangeNumInt;
        maxDuration = remainder === 0 ? maxDuration : maxDuration + (rangeNumInt - remainder);

        const sectionRange = maxDuration / rangeNumInt;

        const resultDurations: number[] = Array(rangeNumInt).fill(0);
        const resultSections: string[] = Array(rangeNumInt).fill("");

        // calculate ranges
        for (let i = 0; i < rangeNumInt; i++) {
            const start = i * sectionRange;
            const end = (i + 1) * sectionRange - 1;
            resultSections[i] = `${start}-${end}s`;
            resultDurations[i] = end;
        }

        this._sectionValues = resultDurations;
        this._labels = resultSections;
    }
```

Reply:
Thats a way better solution. You could also get rid of `this._rangeNum` completely and reference `queryParams.rangeNum` in the template like:

```html
<mdc-select change.delegate="_rangeNumChanged()"
                            class="w50"
                            label="Number of Sections"
                            outlined
                            value.bind="queryParams.rangeNum"
                >
```
then you don't need `this._rangeNum` anymore:
```typescript
const rangeNumInt = parseInt(this.queryParams.rangeNum)
```